File: PDFaCodigoJavaCSharpRustPython/PA_StopWatch/PythonCode/Cstopwatch.py
class SwingTimerEmulator:
    def __init__(self, interval, action, root):
        self.interval = interval
        self.action = action
        self.root = root
        self.timer_id = None
        self.is_running = False

    def start(self):
        if not self.is_running:
            self.is_running = True
            self._tick()

    def stop(self):
        self.is_running = False
        if self.timer_id:
            self.root.after_cancel(self.timer_id)

    def _tick(self):
        if self.is_running:
            self.action()
            self.timer_id = self.root.after(self.interval, self._tick)

class Cstopwatch:
    def __init__(self, model, view):
        self.model = model
        self.view = view
        
        def timer_action():
            self.view.setTime(self.model.getFormattedTime())
            
        self.timer = SwingTimerEmulator(1000, timer_action, self.view)

        def start_action():
            self.model.start()
            self.timer.start()
        self.view.getStartButton().config(command=start_action)

        def stop_action():
            self.model.stop()
            self.timer.stop()
        self.view.getStopButton().config(command=stop_action)

        def exit_action():
            self.view.destroy()
        self.view.getExitButton().config(command=exit_action)

File: PDFaCodigoJavaCSharpRustPython/PA_StopWatch/CSharpCode/Cstopwatch.cs
using System;
using System.Windows.Threading;

namespace CSharpCode {
    public class Cstopwatch {
        private Mstopwatch model;
        private Vstopwatch view;
        private DispatcherTimer timer;

        public Cstopwatch(Mstopwatch model, Vstopwatch view) {
            this.model = model;
            this.view = view;

            timer = new DispatcherTimer();
            timer.Interval = TimeSpan.FromMilliseconds(1000);
            timer.Tick += (sender, e) => {
                view.setTime(model.getFormattedTime());
            };

            view.getStartButton().Click += (sender, e) => {
                model.start();
                timer.Start();
            };

            view.getStopButton().Click += (sender, e) => {
                model.stop();
                timer.Stop();
            };

            view.getExitButton().Click += (sender, e) => {
                System.Windows.Application.Current.Shutdown(); // System.exit(0) equivalente
            };
        }
    }
}